src/Domain/Creators/TimeDependentOptions/FromVolumeFile.cpp

```diff
@@ -0,0 +1,189 @@
+  }
+
+  const auto functions_of_time = deserialize<std::unordered_map<
+      std::string, std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>>(
+      serialized_functions_of_time->data());
+
+  if (not functions_of_time.contains(function_of_time_name)) {
+    PARSE_ERROR(context, "No function of time named "
+                             << function_of_time_name << " in the subfile "
+                             << subfile_name << " of the H5 file "
+                             << h5_filename);
+  }
+
+  const auto& function_of_time = functions_of_time.at(function_of_time_name);
+  const std::array<double, 2> time_bounds = function_of_time->time_bounds();
+
+  if (time < time_bounds[0] or time > time_bounds[1]) {
+    PARSE_ERROR(context, function_of_time_name
+                             << ": The requested time " << time
+                             << " is out of the range of the function of time "
+                             << time_bounds << " from the subfile "
+                             << subfile_name << " of the H5 file "
+                             << h5_filename);
+  }
+
+  return function_of_time->get_clone();
+}
+}  // namespace
+
+template <typename FoTName>
+FromVolumeFile<FoTName>::FromVolumeFile(const std::string& h5_filename,
+                                        const std::string& subfile_name,
+                                        const double time,
+                                        const Options::Context& context) {
+  const std::string function_of_time_name = pretty_type::name<FoTName>();
+
+  const auto function_of_time = get_function_of_time(
+      function_of_time_name, h5_filename, subfile_name, time, context);
+
+  values = function_of_time->func_and_2_derivs(time);
+}
+
+FromVolumeFile<names::Expansion>::FromVolumeFile(
+    const std::string& h5_filename, const std::string& subfile_name,
+    const double time, const Options::Context& context) {
+  const std::string expansion_function_of_time_name{"Expansion"};
+
+  const auto exp_function_of_time =
+      get_function_of_time(expansion_function_of_time_name, h5_filename,
+                           subfile_name, time, context);
+  const auto exp_outer_boundary_function_of_time =
+      get_function_of_time(expansion_function_of_time_name + "OuterBoundary",
+                           h5_filename, subfile_name, time, context);
+
+  expansion_values = exp_function_of_time->func_and_2_derivs(time);
+  expansion_values_outer_boundary =
+      exp_outer_boundary_function_of_time->func_and_2_derivs(time);
+}
+
+FromVolumeFile<names::Rotation>::FromVolumeFile(
+    const std::string& h5_filename, const std::string& subfile_name,
+    const double time, const bool is_quat_fot,
+    const Options::Context& context) {
+  const std::string function_of_time_name{"Rotation"};
+
+  const auto function_of_time = get_function_of_time(
+      function_of_time_name, h5_filename, subfile_name, time, context);
+
+  if (is_quat_fot) {
```

Comment:
You could get rid of this option by checking the result of the dynamic cast. As a user I wouldn't know what value to choose for this option.